### DISCUSSION TITLE
新版本更新预合成的透明度调整无法整体透明

### DISCUSSION THREAD

**Discussion by Johnny87975** [Tencent/libpag#2093] [Q&A] [answered]
[基于最新的更新后，预合成整体调整透明度，预合成无法整体透明，会出现预合成内文件图层重影，这个问题比较严重，导致无法产出，请及时修复或提供历史版本
![WechatIMG474](https://github.com/Tencent/libpag/assets/4196738/c13ada9f-49a4-409f-ac03-f285f9f7f35c)

**Comment by kevingpqi123** [comment]
为了提升对接效率，反馈问题请提供完整的信息：出现问题的版本号、渲染正常的版本号以及复现 demo

**Comment by domchen** [comment, marked as answer]
这个一直都是如此，并不是新版本改了，官网文档里一直也都有注明：https://pag.art/feature.html
<img width="1062" alt="Screenshot 2024-01-18 at 10 05 08" src="https://github.com/Tencent/libpag/assets/6849001/7df99a5b-328b-40f4-9a2a-0e216938bcd8">

这个限制是出于性能优化的原因，如果按照整体应用透明度会导致大量离屏渲染，性能会非常差，现代的用于实时渲染的框架一般都是采用这个策略。建议直接设置对应子元素的透明度。

**Reply by Johnny87975** [reply]
你好，如果是这样的话，没有特别好的解决方案。不离屏对应子元素透明度只要是分层动作都会出现图层重影，预合成的里的元素进行动作在我们的方案里是能进行一些对称性动作，在日常项目里涉及这么干的案例有非常多，相当于把动作整合打组，但是现在这个涉及到透明度调整重影可用性不高，如果可以的话，如果可以的话，希望加上一个开关能够提供是否能进行整体调整透明度，让我们自行决定这么做优化问题。
![1705544611125](https://github.com/Tencent/libpag/assets/4196738/f870af4a-09f6-421c-bb90-251bff9263fe)

**Comment by Johnny87975** [comment]
你好，如果是这样的话，没有特别好的解决方案。不离屏对应子元素透明度只要是分层动作都会出现图层重影，预合成的里的元素进行动作在我们的方案里是能进行一些对称性动作，在日常项目里涉及这么干的案例有非常多，相当于把动作整合打组，但是现在这个涉及到透明度调整重影可用性不高，如果可以的话，如果可以的话，希望加上一个开关能够提供是否能进行整体调整透明度，让我们自行决定这么做优化问题。









逸铭
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Tencent/libpag"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年1月18日(星期四) 上午10:08
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Tencent/libpag] 新版本更新预合成的透明度调整无法整体透明 (Discussion #2093)





 
这个一直都是如此，并不是新版本改了，官网文档里一直也都有注明：https://pag.art/feature.html
 Screenshot.2024-01-18.at.10.05.08.png (view on web)
 
这个限制是出于性能优化的原因，如果按照整体应用透明度会导致大量离屏渲染，性能会非常差，现代的用于实时渲染的框架一般都是采用这个策略。建议直接设置对应子元素的透明度。

**Reply by domchen** [reply]
你说的是确实合理的需求场景，但问题是可能比较小众，实际并没有你说的那么普遍。因为这个设定从六年前就是这样，几乎没有业务提过类似需求或者遇到问题。我们应该不会因为比较小众的需求场景去特别增加API，建议还是探索一下制作上绕过的方式，应该是有比较简单绕过的方法。最简单的一个方式是把需要整体调整透明度的预合成设置为BMP预合成导出。
